src/js/apps/patients/sidebar/filters-sidebar_app.js

```diff
@@ -2,43 +2,44 @@ import Radio from 'backbone.radio';
 
 import App from 'js/base/app';
 
-import { LayoutView, GroupsFilterView } from 'js/views/patients/sidebar/filters/filters-sidebar_views';
+import { LayoutView, CustomFiltersView, groupLabelView } from 'js/views/patients/sidebar/filters/filters-sidebar_views';
 
 export default App.extend({
-  stateEvents: {
-    'change': 'onChangeFilters',
-  },
-  onStart({ state }) {
+  onStart({ state }, directories) {
     this.currentClinician = Radio.request('bootstrap', 'currentUser');
+    this.directories = Radio.request('bootstrap', 'currentOrg:directories');
 
     this.showView(new LayoutView());
 
-    this.showGroupsFilterView();
+    this.showCustomFiltersView();
   },
   viewEvents: {
     'close': 'stop',
     'click:clear:filters': 'onClearFilters',
   },
-  onChangeFilters() {
-    this.showGroupsFilterView();
-  },
-  showGroupsFilterView() {
+  showCustomFiltersView() {
+    const collection = this.directories.clone();
     const groups = this.currentClinician.getGroups();
 
-    const groupsFilterView = new GroupsFilterView({
-      groups,
-      groupId: this.getState('groupId'),
-    });
+    if (groups.length > 1) {
+      const groupDirectory = collection.unshift({
+        name: groupLabelView,
+        slug: 'groupId',
+      });
 
-    this.listenTo(groupsFilterView, 'select:group', this.onSelectGroup);
+      groupDirectory.options = groups;
+    }
 
-    this.showChildView('groups', groupsFilterView);
+    const customFiltersView = new CustomFiltersView({
+      collection,
+      state: this.getState(),
+    });
+
+    this.showChildView('customFilters', customFiltersView);
   },
   onClearFilters() {
-    this.setState('groupId', null);
-  },
-  onSelectGroup(id) {
-    this.setState('groupId', id);
+    this.getState().clear();
+    this.triggerMethod('reset:filters:state');
```

Comment:
If only `this.getState().clear()` is called here, the worklist doesn't restart. And, therefore, the worklist doesn't make a new request to the api with the default filters used. I'm not entirely sure why... it seems to only update the state of the filters in the sidebar, and nothing else.

I added a `this.triggerMethod('reset:filters:state');` here that passes an event up to the `worklist_app.js`, where it resets the `state.filters` to it's default and forces the worklist to restart. Feels overkill, but it was the only way I could get direct access to the state methods.